src/Clb.hs

```diff
@@ -194,50 +185,57 @@ newtype FailReason
   deriving (Show)
 
 -- | Run CLB emulator.
-runClb :: Clb a -> ClbState -> (a, ClbState)
-runClb (ClbT act) = runState act
-
--- | Init emulator state.
-initClb :: MockConfig -> Api.Value -> Api.Value -> ClbState
-initClb
-  cfg@MockConfig {mockConfigProtocol = pparams}
-  _initVal
-  walletFunds =
-    ClbState
-      { emulatedLedgerState = setUtxo pparams utxos (initialState pparams)
-      , mockDatums = M.empty
-      , mockConfig = cfg
-      , mockInfo = mempty
-      , mockFails = mempty
-      }
-    where
-      utxos = L.UTxO $ M.fromList $ mkGenesis walletFunds <$> [1 .. 10]
-
-      -- genesis :: (L.TxIn (Core.EraCrypto EmulatorEra), Core.TxOut EmulatorEra)
-      -- genesis =
-      --   ( L.mkTxInPartial genesisTxId 0
-      --   , L.TxOutCompact
-      --       (L.compactAddr $ mkAddr' $ intToKeyPair 0) -- TODO: use wallet distribution
-      --       (fromJust $ L.toCompact $ C.toMaryValue $ valueToApi initVal)
-      --   )
-
-      mkGenesis :: Api.Value -> Integer -> (L.TxIn (Core.EraCrypto EmulatorEra), Core.TxOut EmulatorEra)
-      mkGenesis walletFund wallet =
-        ( L.mkTxInPartial genesisTxId wallet
-        , L.TxOutCompact
-            (L.compactAddr $ mkAddr' $ intToKeyPair wallet)
-            (fromJust $ L.toCompact $ C.toMaryValue walletFund)
-        )
-
-      mkAddr' payKey = L.Addr L.Testnet (TL.mkCred payKey) L.StakeRefNull
-
-      -- \| genesis transaction ID
-      genesisTxId :: L.TxId L.StandardCrypto
-      genesisTxId = L.TxId $ L.unsafeMakeSafeHash dummyHash
-
-      -- Hash for genesis transaction
-      dummyHash :: Crypto.Hash Crypto.Blake2b_256 Core.EraIndependentTxBody
-      dummyHash = Crypto.castHash $ Crypto.hashWith CBOR.serialize' ()
+runClb :: Clb a -> ClbRuntime -> IO (a, Log LogEntry)
+runClb (Clb act) = runWriterT . runReaderT act
+
+-- Based on: https://github.com/IntersectMBO/cardano-node/blob/master/cardano-testnet/src/Testnet/Property/Run.hs
+-- They are using hedgehog (property testing framework) to orchestrate a testnet running in the background
+-- ....for some god forsaken reason
```

Comment:
I've heard about that. We also have some budget within CTL to integrate cardano-testnet into it instead of Plutip so we can work together here. We have a preliminary agreement with IOG to upstream some changes to cardano-testnet.